test/csuite/timestamp_abort/main.c

```diff
@@ -340,6 +371,241 @@ set_flush_tier_delay(WT_RAND_STATE *rnd)
+    }
+
+    /* Prepare the directory. */
+    testutil_check(__wt_snprintf(backup_home, sizeof(backup_home), "backup.%" PRIu32, index));
+    testutil_system("rm -rf %s && mkdir %s", backup_home, backup_home);
+
+    /* Open the session. */
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Open the backup cursor to get the list of files to copy. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf),
+      "incremental=(src_id=ID%" PRIu32 ",this_id=ID%" PRIu32 ")", src_index, index));
+    testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
+
+    /* Process the files. */
+    while ((ret = cursor->next(cursor)) == 0) {
+        nfiles++;
+        testutil_check(cursor->get_key(cursor, &filename));
+
+        /* Process ranges within each file. */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(file=%s)", filename));
+        testutil_check(session->open_cursor(session, NULL, cursor, buf, &file_cursor));
+
+        first_range = true;
+        rfd = wfd = 0;
+        while ((ret = file_cursor->next(file_cursor)) == 0) {
+            testutil_check(file_cursor->get_key(file_cursor, &offset, &size, &type));
+            testutil_assert(type == WT_BACKUP_FILE || type == WT_BACKUP_RANGE);
+
+            if (type == WT_BACKUP_RANGE) {
+
+                /*
+                 * If this is the first range, copy the base file from the source backup. Then open
+                 * the source and target files to set up the rest of the copy.
+                 */
+                if (first_range) {
+                    first_range = false;
+                    testutil_system(
+                      "cp %s/%s %s/%s", src_backup_home, filename, backup_home, filename);
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", WT_HOME_DIR, filename));
+                    testutil_assert((rfd = open(buf, O_RDONLY, 0666)) >= 0);
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", backup_home, filename));
+                    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+                }
+
+                /* Copy the range. */
+                tmp = dcalloc(1, size);
+                rdsize = pread(rfd, tmp, (size_t)size, (wt_off_t)offset);
+                testutil_assert(rdsize >= 0);
+                testutil_assert(pwrite(wfd, tmp, (size_t)rdsize, (wt_off_t)offset) == rdsize);
+                free(tmp);
+
+                nranges++;
+            } else {
+
+                /* We are supposed to do the full file copy. */
+                testutil_assert(first_range);
+                first_range = false;
+                testutil_system("cp %s/%s %s/%s", WT_HOME_DIR, filename, backup_home, filename);
+            }
+        }
+
+        if (rfd > 0)
+            testutil_assert(close(rfd) == 0);
+        if (wfd > 0)
+            testutil_assert(close(wfd) == 0);
+
+        if (first_range) {
```

Comment:
I think this needs a bigger comment. Something like:
/* 
 * If we get here and first_range is still true it means there there were no changes to the file.
 * The duplicate backup cursor did not return any information. Therefore "copy" the file from
 * the source backup...
 * /